src/vm/types/signatures.rs

```diff
@@ -379,6 +379,31 @@ impl TupleTypeSignature {
     }
 }
 
+impl FunctionSignature {
+
+    pub fn check_args(&self, args: Vec<TypeSignature>) -> bool {
```

Comment:
This should be called something like `check_args_trait_compliance`, because the type admission checks here are specific to determining whether or not a function's arg match a trait.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1246/commits/4d57ca896b7e26c39b02f0bf1b1fc0ff30326169